src/PKSim.Core/Commands/RemoveMoleculeFromPopulationCommand.cs

```diff
@@ -29,8 +30,8 @@ protected override void ClearReferences()
 
       protected override void PerformExecuteWith(IExecutionContext context)
       {
-         _population.RemoveMolecule(_molecule);
-         context.Unregister(_molecule);
+         var removedContainer= _population.RemoveMolecule(_molecule);
+         removedContainer.Each(context.Unregister);
```

Comment:
Probably a way to unfiy those two implementations but at the ned of the day, they are also different use cases so I am leaving it like this for now